dev-tools/releng/src/cmd.rs

```diff
@@ -0,0 +1,167 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::ffi::OsStr;
+use std::path::Path;
+use std::process::ExitStatus;
+use std::process::Output;
+use std::process::Stdio;
+use std::time::Instant;
+
+use anyhow::ensure;
+use anyhow::Context;
+use anyhow::Result;
+use slog::debug;
+use slog::Logger;
+
+/// Wrapper for `tokio::process::Command` where the builder methods take/return
+/// `self`, plus a number of convenience methods.
+pub(crate) struct Command {
+    inner: tokio::process::Command,
+}
+
+impl Command {
+    pub(crate) fn new(program: impl AsRef<OsStr>) -> Command {
+        Command { inner: tokio::process::Command::new(program) }
+    }
+
+    pub(crate) fn arg(mut self, arg: impl AsRef<OsStr>) -> Command {
+        self.inner.arg(arg);
+        self
+    }
+
+    pub(crate) fn args(
+        mut self,
+        args: impl IntoIterator<Item = impl AsRef<OsStr>>,
+    ) -> Command {
+        self.inner.args(args);
+        self
+    }
+
+    pub(crate) fn current_dir(mut self, dir: impl AsRef<Path>) -> Command {
+        self.inner.current_dir(dir);
+        self
+    }
+
+    pub(crate) fn env(
+        mut self,
+        key: impl AsRef<OsStr>,
+        value: impl AsRef<OsStr>,
+    ) -> Command {
+        self.inner.env(key, value);
+        self
+    }
+
+    pub(crate) fn env_remove(mut self, key: impl AsRef<OsStr>) -> Command {
+        self.inner.env_remove(key);
+        self
+    }
+
+    pub(crate) async fn is_success(mut self, logger: &Logger) -> Result<bool> {
+        self.inner
+            .stdin(Stdio::null())
+            .stdout(Stdio::inherit())
+            .stderr(Stdio::inherit());
+        Ok(xtrace(&mut self, logger).await?.status.success())
+    }
+
+    pub(crate) async fn ensure_success(
+        mut self,
+        logger: &Logger,
+    ) -> Result<()> {
+        self.inner
+            .stdin(Stdio::null())
+            .stdout(Stdio::inherit())
+            .stderr(Stdio::inherit());
+        let status = xtrace(&mut self, logger).await?.status;
+        check_status(self, status)
+    }
+
+    pub(crate) async fn ensure_stdout(
+        mut self,
+        logger: &Logger,
+    ) -> Result<String> {
+        self.inner
+            .stdin(Stdio::null())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::inherit());
+        let output = xtrace(&mut self, logger).await?;
+        check_status(self, output.status)?;
+        String::from_utf8(output.stdout).context("command stdout was not UTF-8")
+    }
+
+    pub(crate) fn into_parts(self) -> (Description, tokio::process::Command) {
+        (Description { str: self.to_string() }, self.inner)
+    }
+}
+
+impl std::fmt::Display for Command {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let command = self.inner.as_std();
+        for (name, value) in command.get_envs() {
```

Comment:
How noisy is this? I think this would print out the entire environment, which would be a lot: on my machine, `export | wc -c` returns 4923.

Reply:
It only prints the environment variables explicitly set in the command. https://doc.rust-lang.org/stable/std/process/struct.Command.html#method.get_envs